movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
     for _ in range(order):
         result = result.differentiate("time")
     return result
+
+
+def cdist(
+    a: xr.DataArray,
+    b: xr.DataArray,
+    dim: Literal["individuals", "keypoints"],
+    metric: str | None = "euclidean",
+    **kwargs,
+) -> xr.DataArray:
+    """Compute distance between each pair of the two collections of inputs.
+
+    This function is a wrapper around :func:`scipy.spatial.distance.cdist`
+    and computes the pairwise distances between either a pair of
+    ``individuals`` or ``keypoints`` as specified by ``dim``.
+    The distances are computed using the specified ``metric``.
+
+    Parameters
+    ----------
+    a : xarray.DataArray
+        The first input data containing position vectors of a
```

Comment:
If we add "array" here we will need to do so for all other functions in the module for consistency. But I don't think this is absolutely necessary, since we already mention the types immediately before it.

Reply:
You are right, it's redundant to mention "array" every time, given that the type is `DataArray`. I think what irritated me here is that the phrasing "first input data" implied data is singular, so I thought adding "array" would shift the 'singularity' to array, which is uncontroversial.

Reply:
Now that _cdist is private, I guess this doesn't matter as much.

Reply:
indeed.